### PR TITLE
Fixed static analyzer warnings in SimG4Core/Notification and clean-up Alignment/LaserAlignmentSimulation

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
This PR only fixing static analyzer warnings in SimG4Core/Notification and fix the compilation problem for the new Geant4 11.0 in Alignment/LaserAlignmentSimulation. The last fix is expected to work both for Geant4 10.7 (current CMS default) and for Geant4 11.0. 

Should not affect results.

#### PR validation:
private

### CODE REVIEW

File: Alignment/LaserAlignmentSimulation/src/LaserOpticalPhysicsList.cc [hunk 1]
```diff
@@ -93,7 +93,6 @@ void LaserOpticalPhysicsList::ConstructProcess() {
   pManager->AddDiscreteProcess(theBoundaryProcess);
   pManager->AddDiscreteProcess(theWLSProcess);
 
-  theScintProcess->SetScintillationYieldFactor(1.);
```
Comment (francescobrivio) [inline Alignment/LaserAlignmentSimulation/src/LaserOpticalPhysicsList.cc:96]: @civanch just for my understanding, is this removed due to SA warning? 
If yes, could you link it? thanks a lot!

Comment (civanch) [reply Alignment/LaserAlignmentSimulation/src/LaserOpticalPhysicsList.cc:96]: @francescobrivio , in recent Geant4 this value is 1. by default and in the next Geant4 11.0 this method is removed. I should extend the description of the PR.

**Comment by civanch** [comment]
+1

**Comment by francescobrivio** [comment]
+alca

 - techinical from the alca point of view
 - failure in 4.76 input is unrelated to this PR and shows up in every PR

File: SimG4Core/Notification/src/GenParticleInfoExtractor.cc [hunk 2]
```diff
@@ -3,16 +3,19 @@
...
+                "GenParticleInfoExtractor: G4PrimaryParticle has no user information");
+  } else {
+    GenParticleInfo *gpi = dynamic_cast<GenParticleInfo *>(up);
+    if (gpi == nullptr)
+      G4Exception("SimG4Core/Notification",
+                  "mc001",
+                  FatalException,
+                  "GenParticleInfoExtractor: user information in G4PrimaryParticle is not of GenParticleInfo type");
+  }
   return *gpi;
```
Comment (perrotta) [inline SimG4Core/Notification/src/GenParticleInfoExtractor.cc:20]: Uhm... Now `gpi` is a local variable included in the previous scope.
I wonder what can be returned here

**Review by perrotta — COMMENTED** [review]
Besides the bug pointed out here, that can be easily fixed, this PR does not "fix" any static analyzer warning: they are all still there ("Returning null reference"), and mostly due to the fact that `G4Exception` actually throws, even if it is not known to the static analyzer.
I think we could simply ingnore those SA warnings from `GenParticleInfoExtractor` and `NewTrackAction`, because we know that they are not relevant; while I would concentrate of the possibility of returning a null pointer in `TrackInformationExtractor`, which is not fixed here and if deemed to represent an issue it should be addressed somehow (or also ignored if, in the contrary, returning a null pointer does not bother).

File: SimG4Core/Notification/src/GenParticleInfoExtractor.cc (hunk 2 above)
Comment (makortel) [reply SimG4Core/Notification/src/GenParticleInfoExtractor.cc:20]: There is still `gpi` on this scope on line 6, and line 13 is a duplicate of that, so technically the code does what is intended, but it is confusing. I personally find the earlier version more clear than the changed version.

Comment (civanch) [reply SimG4Core/Notification/src/GenParticleInfoExtractor.cc:18]: Is fixed now

**Comment by civanch** [comment]
please test

**Comment by tvami** [comment]
@civanch could you please include `SimG4Core/Notification` and `Alignment/LaserAlignmentSimulation` in the PR title? The current title is somewhat generic. Thanks!

**Comment by tvami** [comment]
+alca
 * with the remark of renaming the PR

**Comment by civanch** [comment]
+1

**Comment by civanch** [comment]
@perrotta , it seems all concerns are resolved and this PR is OK.

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**